Packs/DuoAdminApi/Integrations/DuoEventCollector/DuoEventCollector_test.py

```diff
@@ -269,3 +406,246 @@ def test_validate_request_order_array(log_type_list, expected_res):
+        "limit": "10",
+        "proxy": False,
+        "retries": "5",
+        "secret_key": {"password": "password", "passwordChanged": False},
+        "end_window": datetime.strptime("2020-01-24 15:11:33", DATE_FORMAT),
+        "fetch_delay": "5"
+    }
+
+    client = Client(Params(**params, mintime={}))
+
+    request_order = ['ADMINISTRATION']
+
+    get_events_obj = GetEvents(client, request_order)
+    output_events, reached_end_window = get_events_obj.events_in_window(input_events)
+    assert len(output_events) == 6
+    assert reached_end_window
+
+
+@freeze_time("2020-01-24 15:16:33 UTC")
+def test_events_in_window_none_in():
+    """ case b
+    Given:
+        A list of log/events.
+    When:
+        calling events_in_window.
+    Then:
+        Validate that none of the events are returned.
+    """
+    input_events = load_json('./test_data/events_in_window_v1_administration.json')
+    params = {
+        "after": "1 month",
+        "host": "api-host.duosecurity.com",
+        "integration_key": "XXXXXXXXXXXXXXXX",
+        "limit": "10",
+        "proxy": False,
+        "retries": "5",
+        "secret_key": {"password": "password", "passwordChanged": False},
+        "end_window": datetime.strptime("2020-01-24 15:09:33", DATE_FORMAT),
+        "fetch_delay": "7"
+    }
+
+    client = Client(Params(**params, mintime={}))
+
+    request_order = ['ADMINISTRATION']
+
+    get_events_obj = GetEvents(client, request_order)
+    output_events, reached_end_window = get_events_obj.events_in_window(input_events)
+    assert len(output_events) == 0
+    assert reached_end_window
+
+
+@freeze_time("2020-01-24 15:16:33 UTC")
+def test_events_in_window_all_no_delay():
+    """ case a
+    Given:
+        A list of log/events.
+    When:
+        calling events_in_window.
+    Then:
+        Validate that all events are returned, because we don't want to apply a delay.
+    """
+    input_events = load_json('./test_data/events_in_window_v1_administration.json')
+    params = {
+        "after": "1 month",
+        "host": "api-host.duosecurity.com",
+        "integration_key": "XXXXXXXXXXXXXXXX",
+        "limit": "10",
+        "proxy": False,
+        "retries": "5",
+        "secret_key": {"password": "password", "passwordChanged": False},
+        "end_window": datetime.strptime("2020-01-24 15:11:33", DATE_FORMAT),
+        "fetch_delay": "0"
+    }
+
+    client = Client(Params(**params, mintime={}))
+
+    request_order = ['ADMINISTRATION']
+
+    get_events_obj = GetEvents(client, request_order)
+    output_events, reached_end_window = get_events_obj.events_in_window(input_events)
+    assert len(output_events) == 9
+    assert not reached_end_window
+
+
+@freeze_time("2020-01-24 15:16:33 UTC")
+def test_calculate_window():
+    """ case a
+    Given:
+        A list of log/events.
+    When:
+        calling events_in_window.
+    Then:
+        Validate that all events are returned, because we don't want to apply a delay.
```

Comment:
You dont call events_in_window here, what are we checking? looks like we are checking that the correct end_window value is set